gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala

```diff
@@ -299,13 +303,20 @@ object ExpressionConverter extends Logging {
             attributeSeq),
           u)
       case b: BinaryExpression =>
+        b match {
+          case arithmetic: BinaryArithmetic if isDecimalArithmetic(arithmetic)
+            && !conf.decimalOperationsAllowPrecisionLoss =>
```

Comment:
@zzcclp Does CH backend support decimalOperationsAllowPrecisionLoss false mode? Default is true

Reply:
@taiyang-li please help to check this, thanks.

Reply:
@jinchengchenghh @zzcclp  CH don't support this config, and its default behavior is disabling precision loss.

Reply:
I also noticed that decimal type promotion is different 
 in CH and spark/hive: 
- spark/hive: https://cwiki.apache.org/confluence/download/attachments/27362075/Hive_Decimal_Precision_Scale_Support.pdf
- CH:  https://clickhouse.com/docs/en/sql-reference/data-types/decimal#operations-and-result-type

Reply:
So I can put this code in gluten-core.
I implement same decimal type promotion in velox arithmetic.